parlai/tasks/triviaqa/agents.py

```diff
@@ -123,13 +133,42 @@ def __init__(self, opt, shared=None):
         super().__init__(opt, shared)
 
 
+class NoEvidenceWikipediaTeacher(WikipediaTeacher):
+    def __init__(self, opt, shared=None):
+        self.no_evidence = True
+        super().__init__(opt, shared)
+
+
 class VerifiedTeacher(MultiTaskTeacher):
     def __init__(self, opt, shared=None):
         opt = copy.deepcopy(opt)
         opt['task'] = 'triviaqa:VerifiedWikipedia,triviaqa:VerifiedWeb'
         super().__init__(opt, shared)
 
 
+class NoEvidenceUnionTeacher(DialogTeacher):
+    def __init__(self, opt, shared=None):
+        if not hasattr(self, 'prefix'):
```

Comment:
what does this prefix do here? where do we set prefix?

Reply:
Iirc. `prefix` is there to get the `verified-` version of the dev data.